pkg/security/ebpf/c/open.h

```diff
@@ -378,6 +378,7 @@ int __attribute__((always_inline)) dr_open_callback(void *ctx, int retval) {
         .syscall.retval = retval,
         .event.async = syscall->async,
         .event.saved_by_ad = syscall->resolver.saved_by_ad,
+        .event.is_activity_dump_sample = syscall->resolver.ad_state == ACTIVITY_DUMP_RUNNING,
```

Comment:
This one is different from the others because file events are submitted to kernel space rate limiting which happen right before the dentry resolution. The event type check you mentioned along with the timeout check are done at that time, and overall the output of all them is resumed in the `syscall->resolver.ad_state` variable:
- ACTIVITY_DUMP_RUNNING means the event was chosen to be part of an activity dump
- NO_ACTIVITY_DUMP means that we don't need to insert this event in a dump

Does it make sense ?

Reply:
:+1: